src/library/scala/collection/mutable/ListBuffer.scala

```diff
@@ -161,8 +161,21 @@ class ListBuffer[A]
   def update(idx: Int, elem: A): Unit = {
     ensureUnaliased()
     if (idx < 0 || idx >= len) throw new IndexOutOfBoundsException(idx.toString)
-    val p = locate(idx)
-    setNext(p, elem :: getNext(p).tail)
```

Comment:
It seems other uses of `getNext` in this class could also be optimized, for example the one in `insert`. Maybe we add some parameters to `setNext` so it knows wheter it has to update `last0` or not?

Reply:
Actually, I’m not sure my fix is correct because I don’t update the `first` member.

Can we postpone this problem to M5 since it’s more an optimization?

Reply:
Yes, agree. Should we close this PR for now?

Reply:
We can.